src/library_pthread.js

```diff
@@ -185,6 +185,24 @@ var LibraryPThread = {
       if (ENVIRONMENT_IS_PTHREAD && _pthread_self()) ___pthread_tsd_run_dtors();
     },
 
+    onThreadExit: function(tb, exitCode) {
```

Comment:
I wonder if this refactoring out of `onThreadExit` could be done NFC (non-functional change)?

Reply:
Do you mean to create separate PR for this?

From what I see in the code ``PThread.runExitHandlers()`` is called from ``threadExit`` and ``threadCancel`` functions. However in one function it is preceded by two calls:
```
// Disable all cancellation so that executing the cleanup handlers won't trigger another JS
// canceled exception to be thrown.
Atomics.store(HEAPU32, (tb + {{{ C_STRUCTS.pthread.canceldisable }}} ) >> 2, 1/*PTHREAD_CANCEL_DISABLE*/);
Atomics.store(HEAPU32, (tb + {{{ C_STRUCTS.pthread.cancelasync }}} ) >> 2, 0/*PTHREAD_CANCEL_DEFERRED*/);
```

Is this done on purpose? If so then I would be thankful for explaining what I'm missing in my understanding regarding purpose and preconditions for these two functions.

Reply:
Ok, now I remember what this change was for. It was about to fix hang in test_pthread_exit_2_2 (as stated in 1st commit of this PR commit message):
```
1. pthread_exit/2-2 test fails sometimes:
   Exit handlers must execute before setting threadStatus to exited
   (1). That's because setting threadStatus to 1 means main thread
   is free to deallocate the thread object. pthread_join is waiting
   for threadStatus to be set to 1.```

However it looks that since that time implementation changed a bit and this tests still hangs with `onThreadExit` refactoring

Reply:
I've checked this more thoroughly and this change is about to unify calling exit handlers when exiting and when cancelling the thread.

Without this change test_pthread_cond_timedwait_2_6 hangs and never returns.